index.txt

```diff
@@ -125,24 +126,44 @@ install_caddy()
 	echo "Downloading Caddy for $caddy_os/$caddy_arch$caddy_arm..."
 	caddy_file="caddy_${caddy_os}_$caddy_arch${caddy_arm}_custom$caddy_dl_ext"
 	caddy_url="https://caddyserver.com/download/$caddy_os/$caddy_arch$caddy_arm?plugins=$caddy_plugins"
+	caddy_asc="https://caddyserver.com/download/$caddy_os/$caddy_arch$caddy_arm/signature?plugins=$caddy_plugins"
 	echo "$caddy_url"
 
+	type -p gpg >/dev/null 2>&1 && gpg=1 || gpg=0
+
 	# Use $PREFIX for compatibility with Termux on Android
-	rm -rf "$PREFIX/tmp/$caddy_file"
+	dl="$PREFIX/tmp/$caddy_file"
+	rm -rf -- "$dl"
 
 	if type -p curl >/dev/null 2>&1; then
-		curl -fsSL "$caddy_url" -o "$PREFIX/tmp/$caddy_file"
+		curl -fsSL "$caddy_url" -o "$dl"
+		((gpg)) && curl -fsSL "$caddy_asc" -o "$dl.asc"
 	elif type -p wget >/dev/null 2>&1; then
-		wget --quiet "$caddy_url" -O "$PREFIX/tmp/$caddy_file"
+		wget --quiet "$caddy_url" -O "$dl"
+		((gpg)) && wget --quiet "$caddy_asc" -O "$dl.asc"
 	else
 		echo "Aborted, could not find curl or wget"
 		return 7
 	fi
 
+	# Verify download
+	if ((gpg)); then
+		caddy_pgp="65760C51EDEA2017CEA2CA15155B6D79CA56EA34"
+		gpg --keyserver ha.pool.sks-keyservers.net --recv-keys $caddy_pgp >/dev/null 2>&1
```

Comment:
I had that first, but it introduces a more fragile process I think, as the signature is shorter, and the keybase another party, and gpg still needs to be invoked to import the key. I think this is cleaner -- unless I misunderstand the process.